[CI] RepositoryAnalysisFailureIT testFailsOnReadError failing

**Build Scans:**
- [elasticsearch-intake #22038 / part2](https://gradle-enterprise.elastic.co/s/cdkdyiekbte2o)
- [elasticsearch-pull-request #68130 / part-2-fips](https://gradle-enterprise.elastic.co/s/cnkl4s3t5lgni)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:snapshot-repo-test-kit:internalClusterTest" --tests "org.elasticsearch.repositories.blobstore.testkit.analyze.RepositoryAnalysisFailureIT.testFailsOnReadError" -Dtests.seed=804A05CAB0B10715 -Dtests.locale=zh-Hant-HK -Dtests.timezone=America/Porto_Acre -Druntime.java=24
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.repositories.blobstore.testkit.analyze.RepositoryAnalysisFailureIT),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!(testFailsOnReadError),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: safeGet: listener was not completed within the timeout
```

**Issue Reasons:**
- [main] 2 failures in test testFailsOnReadError (0.8% fail rate in 245 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -411,9 +411,6 @@ tests:
 - class: org.elasticsearch.packaging.test.DockerTests
   method: test024InstallPluginFromArchiveUsingConfigFile
   issue: https://github.com/elastic/elasticsearch/issues/126936
-- class: org.elasticsearch.repositories.blobstore.testkit.analyze.RepositoryAnalysisFailureIT
-  method: testFailsOnReadError
-  issue: https://github.com/elastic/elasticsearch/issues/127029
 
 # Examples:
 #

--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/analyze/RepositoryAnalysisFailureIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/analyze/RepositoryAnalysisFailureIT.java
@@ -134,7 +134,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.maxBlobSize(ByteSizeValue.ofBytes(10L));
         request.abortWritePermitted(false);
 
-        final CountDown countDown = new CountDown(between(1, request.getBlobCount() / 2));
+        final CountDown countDown = randomRequestCountDown(request);
         blobStore.setDisruption(new Disruption() {
             @Override
             public byte[] onRead(byte[] actualContents, long position, long length) throws IOException {
@@ -158,7 +158,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.abortWritePermitted(false);
         request.rareActionProbability(0.0); // not found on an early read or an overwrite is ok
 
-        final CountDown countDown = new CountDown(between(1, request.getBlobCount() / 2));
+        final CountDown countDown = randomRequestCountDown(request);
 
         blobStore.setDisruption(new Disruption() {
             @Override
@@ -212,7 +212,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         // leads to CI failures. Therefore, we disable rare actions to improve CI stability.
         request.rareActionProbability(0.0);
 
-        final CountDown countDown = new CountDown(between(1, request.getBlobCount() / 2));
+        final CountDown countDown = randomRequestCountDown(request);
 
         blobStore.setDisruption(new Disruption() {
             @Override
@@ -234,9 +234,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.maxBlobSize(ByteSizeValue.ofBytes(10L));
         request.abortWritePermitted(false);
 
-        // requests that create copies count as two blobs. Halving the count ensures that we trigger the disruption
-        // even if every request is a copy
-        final CountDown countDown = new CountDown(between(1, request.getBlobCount() / 2));
+        final CountDown countDown = randomRequestCountDown(request);
 
         blobStore.setDisruption(new Disruption() {
 
@@ -524,6 +522,12 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
 
     private static void assertPurpose(OperationPurpose purpose) {
         assertEquals(OperationPurpose.REPOSITORY_ANALYSIS, purpose);
+    }
+
+    private static CountDown randomRequestCountDown(RepositoryAnalyzeAction.Request request) {
+        // requests that create copies count as two blobs. Halving the count ensures that we trigger the disruption
+        // even if every request is a copy
+        return new CountDown(between(1, request.getBlobCount() / 2));
     }
 
     public static class TestPlugin extends Plugin implements RepositoryPlugin {

--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/analyze/RepositoryAnalysisFailureIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/analyze/RepositoryAnalysisFailureIT.java
@@ -134,7 +134,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.maxBlobSize(ByteSizeValue.ofBytes(10L));
         request.abortWritePermitted(false);
 
-        final CountDown countDown = new CountDown(between(1, request.getBlobCount()));
+        final CountDown countDown = new CountDown(between(1, request.getBlobCount() / 2));
         blobStore.setDisruption(new Disruption() {
             @Override
             public byte[] onRead(byte[] actualContents, long position, long length) throws IOException {
@@ -158,7 +158,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.abortWritePermitted(false);
         request.rareActionProbability(0.0); // not found on an early read or an overwrite is ok
 
-        final CountDown countDown = new CountDown(between(1, request.getBlobCount()));
+        final CountDown countDown = new CountDown(between(1, request.getBlobCount() / 2));
 
         blobStore.setDisruption(new Disruption() {
             @Override
@@ -212,7 +212,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         // leads to CI failures. Therefore, we disable rare actions to improve CI stability.
         request.rareActionProbability(0.0);
 
-        final CountDown countDown = new CountDown(between(1, request.getBlobCount()));
+        final CountDown countDown = new CountDown(between(1, request.getBlobCount() / 2));
 
         blobStore.setDisruption(new Disruption() {
             @Override

--- a/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/analyze/RepositoryAnalysisFailureIT.java
+++ b/x-pack/plugin/snapshot-repo-test-kit/src/internalClusterTest/java/org/elasticsearch/repositories/blobstore/testkit/analyze/RepositoryAnalysisFailureIT.java
@@ -134,7 +134,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.maxBlobSize(ByteSizeValue.ofBytes(10L));
         request.abortWritePermitted(false);
 
-        final CountDown countDown = randomRequestCountDown(request);
+        final CountDown countDown = createDisruptionCountdown(request);
         blobStore.setDisruption(new Disruption() {
             @Override
             public byte[] onRead(byte[] actualContents, long position, long length) throws IOException {
@@ -158,7 +158,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.abortWritePermitted(false);
         request.rareActionProbability(0.0); // not found on an early read or an overwrite is ok
 
-        final CountDown countDown = randomRequestCountDown(request);
+        final CountDown countDown = createDisruptionCountdown(request);
 
         blobStore.setDisruption(new Disruption() {
             @Override
@@ -212,7 +212,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         // leads to CI failures. Therefore, we disable rare actions to improve CI stability.
         request.rareActionProbability(0.0);
 
-        final CountDown countDown = randomRequestCountDown(request);
+        final CountDown countDown = createDisruptionCountdown(request);
 
         blobStore.setDisruption(new Disruption() {
             @Override
@@ -234,7 +234,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         request.maxBlobSize(ByteSizeValue.ofBytes(10L));
         request.abortWritePermitted(false);
 
-        final CountDown countDown = randomRequestCountDown(request);
+        final CountDown countDown = createDisruptionCountdown(request);
 
         blobStore.setDisruption(new Disruption() {
 
@@ -524,7 +524,7 @@ public class RepositoryAnalysisFailureIT extends AbstractSnapshotIntegTestCase {
         assertEquals(OperationPurpose.REPOSITORY_ANALYSIS, purpose);
     }
 
-    private static CountDown randomRequestCountDown(RepositoryAnalyzeAction.Request request) {
+    private static CountDown createDisruptionCountdown(RepositoryAnalyzeAction.Request request) {
         // requests that create copies count as two blobs. Halving the count ensures that we trigger the disruption
         // even if every request is a copy
         return new CountDown(between(1, request.getBlobCount() / 2));

